### PR TITLE
paginate wasm rest api response

### PR DESCRIPTION
**Pull request by LeoGuo621**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [x] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/wasm/keeper/legacy_querier.go [hunk 1]
```diff
@@ -91,13 +93,31 @@ func queryContractState(ctx sdk.Context, bech, queryMethod string, data []byte,
...
 	case QueryMethodContractStateAll:
+		var params types.QueryParamsWithReverse
+		if err := types.ModuleCdc.UnmarshalJSON(data, &params); err != nil {
+			return nil, common.ErrUnMarshalJSONFailed(err.Error())
+		}
+
 		resultData := make([]types.Model, 0)
 		// this returns a serialized json object (which internally encoded binary fields properly)
 		keeper.IterateContractState(ctx, contractAddr, func(key, value []byte) bool {
 			resultData = append(resultData, types.Model{Key: key, Value: value})
```
Comment (xzav3r) [inline x/wasm/keeper/legacy_querier.go:104]: maybe it is better to filter what you want before appending

Comment (LeoGuo621) [reply x/wasm/keeper/legacy_querier.go:104]: LGTM

File: x/wasm/client/rest/query.go [hunk 2]
```diff
@@ -35,8 +39,25 @@ func listCodesHandlerFn(cliCtx clientCtx.CLIContext) http.HandlerFunc {
...
+		}
+		reverse := false
+		reverseStr := r.FormValue("reverse")
+		if reverseStr == "true" {
+			reverse = true
+		}
+		params := types.NewQueryParamsWithReverse(page, limit, reverse)
+		bz, err := cliCtx.Codec.MarshalJSON(params)
+		if err != nil {
+			common.HandleErrorMsg(w, cliCtx, common.CodeMarshalJSONFailed, err.Error())
```
Comment (BananaLF) [inline x/wasm/client/rest/query.go:55]: ErroMsg suggest use rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())

Comment (LeoGuo621) [reply x/wasm/client/rest/query.go:55]: I see.

**Review by BananaLF — APPROVED** [review]

**Comment by LeoGuo621** [comment]
The response format of the rest API modified by this PR is inappropriate, this PR is deprecated.
Please see https://github.com/okex/exchain/pull/2702

**Pull request closed without merging**